### ISSUE TITLE
SSL listener doesn't accept traffic

### ISSUE THREAD

**Issue by kellycampbell** [erlio/vernemq#405]
### Environment

- VerneMQ Version: 1.0.1 (Docker)
- OS: Debian Jessie (docker, kubernetes). Kernel: 4.4.65-k8s
- Erlang/OTP version (if building from source):

I'm setting up a new test environment using the same docker image file, identical settings except for ssl certs, as a known good working environment. Connecting via SSL hangs indefinitely.

### Expected behavior

Connections to the SSL port should exchange certificate information and connection should complete successfully.

### Actual behaviour

Connections hang indefinitely. 

`# openssl s_client -connect localhost:8883
CONNECTED(00000003)

`

netstat shows a recv-Q size of 306, send-Q size of 0, and this stays constant even into the CLOSE_WAIT stage.

`tcp      306      0 127.0.0.1:8883          127.0.0.1:51260         CLOSE_WAIT  -               
`

The broker is working ok on the main tcp port:

`mosquitto_pub -d -h 100.90.147.13 -p 1883 -t test/foo -m {}
Client mosqpub/6505-gwmq-15768 sending CONNECT
Client mosqpub/6505-gwmq-15768 received CONNACK
Client mosqpub/6505-gwmq-15768 sending PUBLISH (d0, q0, r0, m1, 'test/foo', ... (2 bytes))
Client mosqpub/6505-gwmq-15768 sending DISCONNECT
`

Here's the full listener list from netstat -nl
`Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:8883            0.0.0.0:*               LISTEN      -               
tcp        0      0 100.90.147.13:44053     0.0.0.0:*               LISTEN      -               
tcp        0      0 127.0.0.1:8888          0.0.0.0:*               LISTEN      -               
tcp        0      0 100.90.147.13:8888      0.0.0.0:*               LISTEN      -               
tcp        0      0 100.90.147.13:1883      0.0.0.0:*               LISTEN      -               
tcp        0      0 0.0.0.0:18883           0.0.0.0:*               LISTEN      -               
tcp        0      0 0.0.0.0:9100            0.0.0.0:*               LISTEN      -               
tcp        0      0 100.90.147.13:8080      0.0.0.0:*               LISTEN      -               
tcp        0      0 0.0.0.0:4369            0.0.0.0:*               LISTEN      -               
tcp6       0      0 :::4369                 :::*                    LISTEN      -               
`

The log file is attached. Most of the session traffic is from an ELB load balancer health check.

**Comment by kellycampbell** [comment]
[vernemq-log-ssl-hangs-20170607.txt](https://github.com/erlio/vernemq/files/1058591/vernemq-log-ssl-hangs-20170607.txt)

**Comment by kellycampbell** [comment]
[vernemq-log-ssl-hangs-20170607.txt](https://github.com/erlio/vernemq/files/1058598/vernemq-log-ssl-hangs-20170607.txt)

**Comment by kellycampbell** [comment]
[vernemq-log-ssl-hangs-20170607.txt](https://github.com/erlio/vernemq/files/1058603/vernemq-log-ssl-hangs-20170607.txt)

**Comment by larshesel** [comment]
Hmm, this might be related to this bug: https://github.com/erlang/otp/commit/1364c7308e17d43d1a2244e3f2bf11cfec3789ef which apparently is part of OTP 19.3, but the packages we're providing at the moment are built with OTP 19.2.

**Comment by kellycampbell** [comment]
I just checked with wireshark. My Client Hello packet is only 373 bytes. It gets acked by the tcp stack, but nothing after that.

What are some commands I can try in the erlang console if I run `vernemq attach`?

**Comment by kellycampbell** [comment]
I isolated and found the problem. It was a config issue afterall. 

listener.ssl.certfile pointed to a misnamed file.

**Comment by larshesel** [comment]
Oh, ok - great you found it! Maybe we can improve our config validation and throw warnings if the files don't exist?

**Comment by kellycampbell** [comment]
We ran into this again, but it was very weird this time. We had a test environment mqtt server that had been working fine for several months, and then suddenly started having this same behavior (no restarts or anything, TLS port just stopped working). It was VerneMQ v1.2.0.  tcpdump showed the SYN/ACK and a client hello packet + ack and then nothing else was received from mqtt after that.

The cert and ca_roots that were in use don't expire until Sept 2018 (or 2047 for the ca).
Restarting didn't help. Reverting to v1.1.0 and starting with blank /var/lib/vernemq didn't help. 
Generating a new cert didn't help. No changes to the configs were made, only the updated cert.

I didn't see any obvious errors in the logs. Finally we updated to a 1.3.1 nightly and it worked.
